ux.symfony.com/src/Twig/MarkdownExtension.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Twig;
+
+use App\Service\CookBookParser;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Error\RuntimeError;
+use Twig\Error\SyntaxError;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class MarkdownExtension extends AbstractExtension
+{
+    public function __construct(
+        private readonly CookBookParser $cookBookParser,
+        private readonly Environment $twig,
+    ) {
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('markdown', [$this, 'convert'], ['is_safe' => ['html']]),
+        ];
+    }
+
+    /**
+     * @throws RuntimeError
+     * @throws SyntaxError
+     * @throws LoaderError
+     */
+    public function convert(string $content): string
+    {
+        $content = $this->cookBookParser->getContent($content);
+
+        return $this->twig->render('markdown_content.html.twig', ['content' => $content]);
+    }
```

Comment:
This is not a good idea because we lose any performance / caching benefits.

We should simply include the recipe file, and filter it with the existing markdown extension


```twig

{{ include(the_recipe_template)|markdown_to_html }} 

```